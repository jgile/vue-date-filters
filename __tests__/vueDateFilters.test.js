import VueDateFilters from '../src';
import Vue from 'vue/dist/vue.js';
import moment from "moment-timezone";
moment.tz.setDefault('CST');

describe('VueDateFilters', () => {
    Vue.use(VueDateFilters);

    beforeEach(() => {
        //language=HTML
        document.body.innerHTML = `
            <div id="app">
                <h3>Time</h3>
                <table>
                    <tr>
                        <td>time</td>
                        <td>{{ theDate | time }}</td>
                    </tr>
                    <tr>
                        <td>shortTime</td>
                        <td>{{ theDate | shortTime }}</td>
                    </tr>
                    <tr>
                        <td>longTime</td>
                        <td>{{ theDate | longTime }}</td>
                    </tr>
                </table>
                <br/>

                <h3>Date</h3>
                <table>
                    <tr>
                        <td>date</td>
                        <td>{{ theDate | date }}</td>
                    </tr>
                    <tr>
                        <td>shortDate</td>
                        <td>{{ theDate | shortDate }}</td>
                    </tr>
                    <tr>
                        <td>writtenDate</td>
                        <td>{{ theDate | writtenDate }}</td>
                    </tr>
                </table>
                <br/>

                <h3>DateTime</h3>
                <table>

                    <tr>
                        <td>dateTime</td>
                        <td>{{ theDate | dateTime }}</td>
                    </tr>
                    <tr>
                        <td>shortDateTime</td>
                        <td>{{ theDate | shortDateTime }}</td>
                    </tr>
                    <tr>
                        <td>writtenDateTime</td>
                        <td>{{ theDate | writtenDateTime }}</td>
                    </tr>
                </table>


                
                <h3>Other</h3>
                <table>
                    <tr>
                        <td>utcToLocal</td>
                        <td>{{ theDate | utcToLocal }}</td>
                    </tr>
                    <tr>
                        <td>localToUtc</td>
                        <td>{{ theDate | localToUtc }}</td>
                    </tr>
                </table>
            </div>
        `;
    });

    it('can mount', async () => {
        await createVm();

        expect(document.body.innerHTML).toMatchSnapshot();
    });
});

async function createVm() {
    const vm = new Vue({
        el: '#app',
        data() {
            return {
                theDate: "2018-04-05"
            };
        }
    });

    await Vue.nextTick(() => {
    });

    return {app: vm, component: vm.$children[0]};
}
