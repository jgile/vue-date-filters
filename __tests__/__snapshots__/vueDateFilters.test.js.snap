// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExampleComponent can mount 1`] = `
"
            <div id=\\"app\\"><h3>Time</h3> <table><tbody><tr><td>time</td> <td>12:00 am</td></tr> <tr><td>shortTime</td> <td>12:00</td></tr> <tr><td>longTime</td> <td>12:00:00 am</td></tr></tbody></table> <br> <h3>Date</h3> <table><tbody><tr><td>date</td> <td>2018-04-05</td></tr> <tr><td>shortDate</td> <td>2018-4-5</td></tr> <tr><td>writtenDate</td> <td>Apr 5th, 2018</td></tr></tbody></table> <br> <h3>DateTime</h3> <table><tbody><tr><td>dateTime</td> <td>2018-04-05 12:00 am</td></tr> <tr><td>shortDateTime</td> <td>2018-4-5 12:00 am</td></tr> <tr><td>writtenDateTime</td> <td>Apr 5th, 2018, 12:00 am</td></tr></tbody></table> <h3>Other</h3> <table><tbody><tr><td>utcToLocal</td> <td>\\"2018-04-05T00:00:00.000Z\\"</td></tr> <tr><td>localToUtc</td> <td>\\"2018-04-05T05:00:00.000Z\\"</td></tr></tbody></table></div>
        "
`;
